internal/pkg/siderolink/provision.go

```diff
@@ -345,14 +378,14 @@ func (h *ProvisionHandler) removePendingMachine(ctx context.Context, pendingMach
 	return nil
 }
 
-func establishLink[T res](ctx context.Context, logger *zap.Logger, st state.State, provisionContext *provisionContext, annotations ...string) (*pb.ProvisionResponse, error) {
-	link, err := createResource[T](ctx, st, provisionContext, annotations...)
+func establishLink[T res](ctx context.Context, logger *zap.Logger, st state.State, provisionContext *provisionContext, callbacks ...func(resource.Resource) error) (*pb.ProvisionResponse, error) {
```

Comment:
Should we make this code less smart/generic? Something like `addAnnotations, removeAnnotations []string` as last 2 args? And same for `createResource`?

Reply:
sure, if you think it's better

Reply:
rgr